### PR TITLE
Roll engine to 4c79e423dc6f89f98d8ceb263a5ca78e2f2da996

### PR DESCRIPTION
**Pull request by jason-simmons**
Also includes
  * Updates to affected tests
  * Change flutter_tools to pass package URIs to the Dart front end
    instead of filesystem paths

### CODE REVIEW

**Review by dnfield — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/compile.dart [hunk 1]
```diff
@@ -168,7 +207,17 @@ class KernelCompiler {
...
     if (extraFrontEndOptions != null)
       command.addAll(extraFrontEndOptions);
-    command.add(mainPath);
+
+    Uri mainUri;
+    if (packagesPath != null) {
+      mainUri = _PackageUriMapper.findUri(mainPath, packagesPath);
+    }
+    if (mainUri != null) {
+      command.addAll(<String>['--packages', packagesPath, mainUri.toString()]);
```
Comment (aam) [inline packages/flutter_tools/lib/src/compile.dart]: Not sure why we only pass packagesPath if we were able to resolve mainUri? Conceptually it feels that .packages file is useful even if main file is identified via file path.

File: packages/flutter_tools/lib/src/compile.dart [hunk 2]
```diff
@@ -481,7 +537,7 @@ class ResidentCompiler {
         }
       }
     }
-    return fileUri;
+    return null;
```
Comment (aam) [inline packages/flutter_tools/lib/src/compile.dart]: Not sure why we are returning null if fileSystemRoots is null and we can't map file to package:-uri

File: packages/flutter_tools/lib/src/compile.dart [hunk 3]
```diff
@@ -302,15 +351,22 @@ class ResidentCompiler {
...
+    if (request.packagesFilePath != null) {
+      packageUriMapper = _PackageUriMapper(request.mainPath, request.packagesFilePath);
+    }
+
     if (_server == null) {
-      return _compile(_mapFilename(request.mainPath),
-          request.outputPath, _mapFilename(request.packagesFilePath));
+      return _compile(
+          _mapFilename(packageUriMapper, request.mainPath), request.outputPath,
+          _mapFilename(null, request.packagesFilePath)
```
Comment (aam) [inline packages/flutter_tools/lib/src/compile.dart]: can we decorate null with parameter name? `/* packageUriMapper= */ null`, for example?

File: packages/flutter_tools/lib/src/compile.dart (hunk 1 above)
Comment (jason-simmons) [reply packages/flutter_tools/lib/src/compile.dart]: done

File: packages/flutter_tools/lib/src/compile.dart (hunk 2 above)
Comment (jason-simmons) [reply packages/flutter_tools/lib/src/compile.dart]: This is a helper function wrapped by `mapFilename` and `mapFileUri`.  Returning null tells the wrapper to return the original filename or URI

File: packages/flutter_tools/lib/src/compile.dart (hunk 3 above)
Comment (jason-simmons) [reply packages/flutter_tools/lib/src/compile.dart]: done

**Comment by aartbik** [comment]
yeah!   \O/  \O/  \O/  \O/  \O/  \O/

**Pull request merged**